plutus-scb-client/src/View/Events.purs

```diff
@@ -0,0 +1,129 @@
+import Halogen.HTML (HTML, b_, code_, div_, h2_, span, span_, text)
+import Plutus.SCB.Events (ChainEvent(..))
+import Prelude
+import Bootstrap (badge, badgePrimary, cardBody_, cardHeader_, card_, nbsp)
+import Bootstrap.Extra (preWrap_)
+import Data.Array as Array
+import Data.Foldable.Extra (countConsecutive)
+import Data.Json.JsonMap (JsonMap(..))
+import Data.Lens (to, traversed)
+import Data.Lens.Extra (toArrayOf)
+import Data.String.Extra (abbreviate)
+import Halogen.HTML.Properties (classes)
+import Ledger.Index (UtxoIndex)
+import Ledger.Tx (Tx)
+import Ledger.TxId (TxId)
+import Plutus.SCB.Events.Contract (ContractEvent, ContractInstanceState(..))
+import Plutus.SCB.Events.Node (NodeEvent)
+import Plutus.SCB.Events.User (UserEvent(..))
+import Plutus.SCB.Events.Wallet (WalletEvent)
+import Plutus.SCB.Types (ContractExe(..))
+
+transactionPane ::
+  forall p i.
+  JsonMap TxId Tx -> HTML p i
+transactionPane (JsonMap txMap) =
+  card_
+    [ cardHeader_
+        [ h2_ [ text "Txs" ]
+        ]
+    , cardBody_
+        ( toArrayOf
+            ( traversed
+                <<< to (\x -> div_ [ code_ [ text $ show x ] ])
+            )
+            txMap
+        )
+    ]
+
+utxoIndexPane :: forall p i. UtxoIndex -> HTML p i
+utxoIndexPane utxoIndex =
+  card_
+    [ cardHeader_
+        [ h2_ [ text "UtxoIndex" ]
+        ]
+    , cardBody_ [ div_ [ code_ [ text $ show utxoIndex ] ] ]
+    ]
+
+eventsPane :: forall p i. Array (ChainEvent ContractExe) -> HTML p i
+eventsPane events =
+  card_
+    [ cardHeader_
+        [ h2_ [ text "Event log" ]
+        , text (show (Array.length events))
+        , nbsp
+        , text "Event(s)"
+        ]
+    , cardBody_ [ div_ (countedEventPane <$> countConsecutive events) ]
+    ]
+
+countedEventPane :: forall p i. Int /\ ChainEvent ContractExe -> HTML p i
+countedEventPane (count /\ event) =
+  div_
+    [ preWrap_
+        [ span [ classes [ badge, badgePrimary ] ]
+            [ text $ show count <> "x" ]
+        , nbsp
+        , showEvent event
+        ]
+    ]
+
+showEvent :: forall p i. ChainEvent ContractExe -> HTML p i
+showEvent (ContractEvent subevent) =
+  span_
+    [ b_
+        [ text "Contract:"
+        , nbsp
+        ]
+    , showContractEvent subevent
+    ]
+
+showEvent (UserEvent subevent) =
+  span_
+    [ b_
+        [ text "User:"
+        , nbsp
+        ]
+    , showUserEvent subevent
+    ]
+
+showEvent (WalletEvent subevent) =
+  span_
+    [ b_
+        [ text "Wallet:"
+        , nbsp
+        ]
+    , showWalletEvent subevent
+    ]
+
+showEvent (NodeEvent subevent) =
+  span_
+    [ b_
+        [ text "Node:"
+        , nbsp
+        ]
+    , showNodeEvent subevent
+    ]
+
+showUserEvent :: forall p i. UserEvent ContractExe -> HTML p i
+showUserEvent (InstallContract (ContractExe { contractPath })) = text $ "Install " <> contractPath
+
+showUserEvent ( ContractStateTransition
+    ( ContractInstanceState
+      { csContract
+    , csCurrentState
```

Comment:
do you use purty?

Reply:
Yes. Must be a purty issue, as this is the result. 🤷